lib/active_record/virtual_attributes/virtual_total.rb

```diff
@@ -75,25 +75,25 @@ def define_virtual_aggregate_attribute(name, relation, method_name, column, opti
         end
       end
 
+      def define_virtual_size_method(name, relation)
+        define_method(name) do
+          (attribute_present?(name) ? self[name] : nil) || send(relation).try(:size) || 0
+        end
+      end
+
       def define_virtual_aggregate_method(name, relation, method_name, column)
```

Comment:
Could this technically be a problem if someone passes `:size` to `virtual_aggregate`?  I am not against the change, but it does seem to make it a bit if a breaking change since there is a functional difference as a result.

Reply:
ok.

we own all the callers to this functionality.

55 calls are to `virtual_total`
6 calls are to `virtual_aggregate`, all of which are for `:sum`.

Do we want to ignore this case, or add an escape clause?

```ruby
  def define_virtual_aggregate_method(name, relation, method_name, column)
    # escape clause
    return define_virtual_total(name, relation, options) if method_name == :size

    define_virtual_aggregate_method(name, relation, method_name, column)
    define_virtual_aggregate_attribute(name, relation, method_name, column, options)
 end
```

Reply:
@NickLaMuro I'm leaning towards just ignoring that case / aka let the case that we never call break.

Reply:
Preface:  I don't have merge rights on this project or really have much of a say in it's direction, so this is mostly an opinion response.

(also, sorry I missed your comment yesterday.  oops!  wasn't ignoring you)

> @NickLaMuro I'm leaning towards just ignoring that case / aka let the case that we never call break.

The key part of your argument here is based on "we" (as I read:  the "ManageIQ org"), but this project is called **`activerecord`**`-virtual_attributes`, not **`manageiq`**`-virtual_attributes`, so a project "owned by use" but not "only for us".  To me that gives assumption that this is a project with a some sort of stable API, especially since we are now in a [`1.x`+ version](https://github.com/ManageIQ/activerecord-virtual_attributes/blob/80f8e43d0dd6f4d335140b3a50c1c9f3a3813da1/lib/active_record/virtual_attributes/version.rb#L3).

When you say:

> we own all the callers to this functionality.

You are talking about `github.com/ManageIQ`, not "anyone that could be using this library ever" and I think it is a pretty bad precedence to set that you are will to just breaking functionality on a whim with this project that I see as being designed for others to use.  I am willing to entertain the idea that for the time being we are probably the only users of project (hard to know based off the download numbers from [rubygems.org](https://rubygems.org/gems/activerecord-virtual_attributes)), but I think you and who ever else you consider an owner of this project should determine when this project is no longer considered "beta" enough (if it isn't already) to have a more stable API and ridged depreciation policy.  That, or determine if this really is a project that is intended for others to use, or something we have just broken out (possibly consider a rename if that is the case).

Also, to answer your other question:

> Do we want to ignore this case, or add an escape clause?

I think this would work, but if your end goal is to remove this, I would consider marking this use case as deprecated and remove it in the `2.0` release as well.

Reply:
ok, reving the major version works for me.

The api for this has never been nailed down. So it is hard to know when something changes the public api when we haven't done a good enough job documenting it.

your are right. this should probably be version `0.1.0`

Reply:
Thanks Keenan.  I am good with this solution.

Reply:
> Preface: I don't have merge rights on this project or really have much of a say in it's direction, so this is mostly an opinion response.

I don't know, your opinion on this was pretty valuable to me.  I agree.  Do we want to get the rest of the open PRs in first or do we want this in with them all and just release a minor version bump?

Reply:
@jrafanie I think what we decided (side conversation) was that the fix he put in for now:

```diff
+          return define_virtual_total(name, relation, options) if method_name == :size
```

Will work for the time being to keep functionality the same, and if we want to consider deprecating this and removing it in 2.0, I think we should follow that up in another PR.

Again, I agree with the direction, just not doing it in a "wild west" manner.